thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -132,5 +137,17 @@ private static void populateMergedResult(MergedAnomalyResultDTO mergedAnomaly,
     mergedAnomaly.setStartTime(currentResult.getStartTime());
     mergedAnomaly.setEndTime(currentResult.getEndTime());
     mergedAnomaly.setCreatedTime(System.currentTimeMillis());
+    // populate current result's property as well, will be used to identify if two anomalies can be merged by comparing their property
+    mergedAnomaly.setProperties(AnomalyDetectionUtils.decodeCompactedPropertyStringToMap(currentResult.getProperties()));
+  }
+
+  // compare if two anomalies have same property when doing anomaly detection, if from same detection configuration then is mergeable
+  private static boolean isMergeable(MergedAnomalyResultDTO anomaly1, MergedAnomalyResultDTO anomaly2){
+    // If both of anomalies don't have key TEST_CONFIG_TAG, they are mergeable;
```

Comment:
This is a very dangerous black magic: No one would know isMergeable is enabled unless there is a magic keyword in the properties of anomaly result.

I would create a new method in anomaly function: List<String> getTestIsMergeableProperties(), which returns a list of keys that could be used to retrieve the properties for determining isMergeable. If an anomaly function returns an empty list, then this function's anomalies could be mergeable regardless their properties; otherwise, we compare all values of the test keys.

Please refine the design of this method.